engine/src/main/java/org/entando/entando/web/common/interceptor/EntandoOauth2Interceptor.java

```diff
@@ -0,0 +1,117 @@
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.method.HandlerMethod;
+import org.springframework.web.servlet.handler.HandlerInterceptorAdapter;
+
+/**
+ *
+ * @author paddeo
+ */
+public class EntandoOauth2Interceptor extends HandlerInterceptorAdapter {
+
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+
+    @Autowired
+    private IApiOAuth2TokenManager oAuth2TokenManager;
+
+    @Autowired
+    private IAuthenticationProviderManager authenticationProviderManager;
+
+    @Autowired
+    private IAuthorizationManager authorizationManager;
+
+    @Override
+    public boolean preHandle(HttpServletRequest request, HttpServletResponse response, Object handler) throws Exception {
+        if (handler instanceof HandlerMethod) {
+            HandlerMethod method = (HandlerMethod) handler;
+            if (method.getMethod().isAnnotationPresent(RequestMapping.class)) {
+                RestAccessControl rqm = method.getMethodAnnotation(RestAccessControl.class);
+                String permission = rqm.permission();
+                this.extractOAuthParameters(request, permission);
+            }
+        }
+        return true;
+    }
+
+    protected void extractOAuthParameters(HttpServletRequest request, String permission) {
+        try {
+            logger.info("Permission required: {}", permission);
+            OAuthAccessResourceRequest requestMessage = new OAuthAccessResourceRequest(request, ParameterStyle.HEADER);
+
+            String accessToken = requestMessage.getAccessToken();
+            if (StringUtils.isBlank(accessToken)) {
+                throw new EntandoTokenException("no access token found", request, null);
+            }
+
+            final OAuth2Token token = oAuth2TokenManager.getApiOAuth2Token(accessToken);
+            this.validateToken(request, accessToken, token);
+
+            String username = token.getClientId();
+            this.checkAuthorization(username, permission, request);
+
+        } catch (OAuthSystemException | ApsSystemException | OAuthProblemException ex) {
+            logger.error("System exception {}", ex.getMessage());
+            throw new RestServerError("error parsing OAuth parameters", ex);
+        }
+    }
+
+    protected void checkAuthorization(String username, String permission, HttpServletRequest request) throws ApsSystemException {
+        UserDetails user = authenticationProviderManager.getUser(username);
+        if (user != null) {
+            logger.info("User {} requesting resource that requires {} permission ", username, permission);
+            if (StringUtils.isNotBlank(permission)) {
+                if (!authorizationManager.isAuthOnPermission(user, permission)) {
+                    throw new EntandoAuthorizationException(null, request, username);
+                }
+            }
+        } else {
+            logger.info("User {} not found ", username);
```

Comment:
Are we sure this should be a no-op?

Reply:
Yes. It's the case when an access token represents an API consumer but not an Entando user, so we should not deny access.